src/core/CCList.ml

```diff
@@ -1109,6 +1109,17 @@ let remove_at_idx i l0 =
   remove_at_idx 5 [1;2;3;4] = [1;2;3;4]
 *)
 
+let rec remove_last = function
+  | [] | [_] -> []
+  | x::xs -> x::remove_last xs
```

Comment:
this is not tail-recursive, I'm a bit uncomfortable with it. Can you write something like `remove_at_idx`, or, even better, like `take` (with a fast path and a safe path)?